src/spetlr/etl/loaders/simple_loader.py

```diff
@@ -15,20 +15,29 @@ def append(self, df: DataFrame) -> None:
         pass
 
 
+class Upsertable(Protocol):
+    def upsert(self, df: DataFrame, join_cols: List[str]) -> Union[DataFrame, None]:
+        pass
+
+
 class SimpleLoader(Loader):
     def __init__(
         self,
-        handle: Union[Overwritable, Appendable],
+        handle: Union[Overwritable, Appendable, Upsertable],
         *,
         mode: str = "overwrite",
+        join_cols: List[str] = None,
         dataset_input_keys: Union[str, List[str]] = None,
     ):
         super().__init__(dataset_input_keys=dataset_input_keys)
         self.mode = mode
```

Comment:
Suggestion: we could do `self.mode = mode.lower()` for making the code more robust